File: packages/plugin-simplicity/fixtures/python/archive.py
import errno


def read_archive(path):
    try:
        with open(path, "rb") as handle:
            return handle.read()
    except OSError as error:
        if is_missing(error):
            return b""
        raise


# An exact match: the code is copied character for character from store.py, and only the
# docstring was reworded.
def is_missing(error):
    """True when the path does not exist on disk."""
    return error.errno == errno.ENOENT

File: packages/plugin-simplicity/fixtures/ts/archive.ts
import { rm } from 'node:fs/promises'

export async function dropArchive(path: string): Promise<boolean> {
  try {
    await rm(path)

    return true
  }
  catch (error) {
    if (isNodeError(error) && error.code === 'ENOENT') {
      return false
    }

    throw error
  }
}

// The exact-match twin of `store.ts`: copied character for character.
function isNodeError(error: unknown): error is NodeJS.ErrnoException {
  return error instanceof Error && 'code' in error
}
